conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -183,6 +187,20 @@ private static void frameworkException(HttpServerExchange exchange, FrameworkExc
         writeResponse(exchange, Optional.of(ConjureError.fromServiceException(exception)), statusCode);
     }
 
+    private static void deadlineExpiredException(HttpServerExchange exchange, DeadlineExpiredException exception) {
+        // need to further narrow down to external versus internal exception types
+        if (exception instanceof DeadlineExpiredException.External externalException) {
+            log.info("external deadline exceeded", externalException);
+            writeResponse(exchange, Optional.empty(), 400);
+        } else if (exception instanceof DeadlineExpiredException.Internal internalException) {
+            log.error("internal deadline exceeded", internalException);
+            writeResponse(exchange, Optional.empty(), 500);
```

Comment:
I think it's probably less important for callers to distinguish between whether an internal deadline has been exceeded vs other types of 500, but if we can convey it, we may be able to make the system a bit easier to observe.